lib/models/application.coffee

```diff
@@ -237,24 +257,25 @@ getApplicationModel = (deps, opts) ->
 	# @function
 	# @memberof resin.models.application
 	#
-	# @param {String} name - application name
+	# @param {String|Number} nameOrId - application name (string) or id (number)
 	# @returns {Promise}
 	#
 	# @example
 	# resin.models.application.remove('MyApp');
 	#
 	# @example
+	# resin.models.application.remove(123);
+	#
+	# @example
 	# resin.models.application.remove('MyApp', function(error) {
 	# 	if (error) throw error;
 	# });
 	###
-	exports.remove = (name, callback) ->
-		exports.get(name).then ->
+	exports.remove = (nameOrId, callback) ->
+		exports.get(nameOrId).then (application) ->
```

Comment:
Yep, see my original PR notes above. Sounds like there isn't a good reason for this then - I'll remove it.

I'm going to do the PineJS options part as a second PR, because this one is enormous enough already.